### ISSUE TITLE
Always redirected to signup page after a clean install 12.2

### ISSUE THREAD

**Issue by adams79** [strapi/strapi#1235]
**Informations**
- node 9.3.0
- 5.2.0
- 3.0.0-alpha 12.2
- Mongo
- MacOS High Sierra

**What is the current behavior?**
After a clean install, accessing the admin screen I’m always redirected to signup also after registering the new user. I see that the user is registered in Mongo

**Steps to reproduce the problem**
Install Strapi from npm
Create new project
Try to login to the admin and create a new user.
Try to access from another browser window 

**What is the expected behavior?**
After creating the first user, accessing the admin should redirect to signin

**Comment by derrickmehaffy** [comment]
Seems this issue is related to some kind of slowness on the server or client side where a timeout fails and redirects. I've noticed this issue as well.

**Comment by raulriera** [comment]
More info can be found here https://strapi.slack.com/archives/C0BLB2VJ7/p1526608909000243

**Comment by abdonrd** [comment]
Same issue here!

**Comment by johanbaath** [comment]
@adams79 @raulriera this issue is fixed in my pull request which was merged: https://github.com/strapi/strapi/pull/1222

It's not included in 12.2 unfortunately!

**Comment by mnlbox** [comment]
@derrickmehaffy what's the meaning of below sentence when we run strapi on localhost?
> slowness on the server or client side

I don't undrestand :thinking:

**Comment by derrickmehaffy** [comment]
@mnlbox there is some slowdown on either side. Most likely client browser slowness, if something doesn't respond in a certain amount of time (we are talking milliseconds here) then Strapi is redirecting to the register instead of `auth/login`

Personally @soupette I think the init thing should be auto-redirected and instead a link should be provided to register on the login page. Alternatively for the initial user, strapi should check the database if `x` users > 0 than it should always direct to the login page. A simple query to check the database for existing users would work.

**Comment by blaasvaer** [comment]
@raulriera clicking your link sends me to slack (surprise!). Here I'm asked to sign in ...

Now, I obviously don't have an account for strapi.slack.com ... and my only option is this:

`If you have an @strapi.io email address, you can create an account.`

I then did that ... erhm, and then nothing happened, of course.

Now, my question is: Is it automatically assumed that everyone reading issues on this github repo has a @strapi.io e-mail? Or am I totally missing something obvious, having a: ask-comment-click-link-to-slack-be-sent-to-slack-to-just-blankly-stare-at-the-screen-having-absolutely-no-clue-what-to-do-next-to-access-the-link-clicked-moment?

**Comment by derrickmehaffy** [comment]
@blaasvaer https://github.com/strapi/strapi right on the front page in the readme:

![image](https://user-images.githubusercontent.com/8593673/40419697-fa531bd2-5e3a-11e8-9477-3f4289f7a3a0.png)

**Comment by derrickmehaffy** [comment]
And yes, anyone can create a slack account and join the strapi slack. The Slack server was probably created before the community could join it.

**Comment by blaasvaer** [comment]
> And yes, anyone can create a slack account and join the strapi slack. The Slack server was probably created before the community could join it.

Erhm, I didn't get that. What am I supposed to do to create an account for strapi.slack.com?

**Comment by derrickmehaffy** [comment]
@blaasvaer http://lmgtfy.com/?iie=1&q=create+an+account+on+slack

**Comment by blaasvaer** [comment]
Great, having to google how to create an account. That's seriously fucked up UX ; ) ...

**Comment by derrickmehaffy** [comment]
@blaasvaer Considering Slack is not developed, nor controlled by Strapi, I would recommend talking to the Slack devs on that. Though it seems pretty self explanatory to me. I'm sure if you have used an IRC before, the experience will be similar.

**Comment by blaasvaer** [comment]
I know you don't have anything to do with Slack (other than having chosen it as your primary support platform ... now, that could be questioned as being a good idea). And in general it's all the assumptions you guys make ... that's really worrying.

I've done all in my power to access strapi.slack.com but haven't gotten anywhere nearer a solution ... so I think I'll just leave it at that. Thanks.

**Comment by derrickmehaffy** [comment]
@blaasvaer I'm not apart of the Strapi team, just another dev here to help support them and use their project. But since you can't seem to figure out how to join slack here is that convo:

![image](https://user-images.githubusercontent.com/8593673/40420294-d4a67788-5e3c-11e8-90fc-4a0b8ad81068.png)

**Comment by derrickmehaffy** [comment]
@blaasvaer Hence the bug report, its a known issue. Fixes are being worked on, as @soupette has said in the other issues thats linked here and in your bug report there is a PR to fix it. It just hasn't been merged into a release yet.

**Comment by blaasvaer** [comment]
OK, thanks for the update. I'll hold my horses on this project until it reaches a more erhm, mature state. ; )

**Comment by derrickmehaffy** [comment]
Hence the alpha, it's not even in beta yet. :P

**Comment by soupette** [comment]
@blaasvaer you need to create a new app and apply these modifications into /admin/admin/src/containers/AdminPage/index.js

```js
/*
 * AdminPage
 *
 * This is the first thing users see of our AdminPage, at the '/' route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a neccessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import ReactGA from 'react-ga';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Switch, Route } from 'react-router-dom';
import { get, includes, isFunction, map, omit } from 'lodash';
import { compose } from 'redux';

// Actions required for disabling and enabling the OverlayBlocker
import {
  disableGlobalOverlayBlocker,
  enableGlobalOverlayBlocker,
} from 'actions/overlayBlocker';

import { pluginLoaded, updatePlugin } from 'containers/App/actions';
import {
  makeSelectBlockApp,
  makeSelectShowGlobalAppBlocker,
  selectHasUserPlugin,
  selectPlugins,
} from 'containers/App/selectors';

import { hideNotification } from 'containers/NotificationProvider/actions';

// Design
import ComingSoonPage from 'containers/ComingSoonPage';
import Content from 'containers/Content';
import LocaleToggle from 'containers/LocaleToggle';
import CTAWrapper from 'components/CtaWrapper';
import Header from 'components/Header/index';
import HomePage from 'containers/HomePage/Loadable';
import InstallPluginPage from 'containers/InstallPluginPage/Loadable';
import LeftMenu from 'containers/LeftMenu';
import ListPluginsPage from 'containers/ListPluginsPage/Loadable';
import Logout from 'components/Logout';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import OverlayBlocker from 'components/OverlayBlocker';
import PluginPage from 'containers/PluginPage/Loadable';

// Utils
import auth from 'utils/auth';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';

import { getGaStatus, getLayout } from './actions';
import reducer from './reducer';
import saga from './saga';
import selectAdminPage from './selectors';

import styles from './styles.scss';

const PLUGINS_TO_BLOCK_PRODUCTION = ['content-type-builder', 'settings-manager'];

export class AdminPage extends React.Component { // eslint-disable-line react/prefer-stateless-function
  state = { hasAlreadyRegistereOtherPlugins: false };

  getChildContext = () => (
    {
      disableGlobalOverlayBlocker: this.props.disableGlobalOverlayBlocker,
      enableGlobalOverlayBlocker: this.props.enableGlobalOverlayBlocker,
      plugins: this.props.plugins,
      updatePlugin: this.props.updatePlugin,
    }
  );

  componentDidMount() {
    this.checkLogin(this.props);
    this.props.getGaStatus();
    this.props.getLayout();
    ReactGA.initialize('UA-54313258-9');
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.location.pathname !== this.props.location.pathname) {
      this.checkLogin(nextProps);

      if (nextProps.adminPage.allowGa) {
        ReactGA.pageview(nextProps.location.pathname);
      }
    }

    if (get(nextProps.plugins.toJS(), ['users-permissions', 'hasAdminUser']) !== get(this.props.plugins.toJS(), ['users-permissions', 'hasAdminUser'])) {
      this.checkLogin(nextProps, true);
    }

    if (!this.hasUserPluginLoaded(this.props) && this.hasUserPluginLoaded(nextProps)) {
      this.checkLogin(nextProps);
    }
  }

  checkLogin = (props, skipAction = false) => {
    if (props.hasUserPlugin && this.isUrlProtected(props) && !auth.getToken()) {
      if (!this.hasUserPluginLoaded(props)) {
        return;
      }

      const endPoint = this.hasAdminUser(props) ? 'login': 'register';
      this.props.history.push(`/plugins/users-permissions/auth/${endPoint}`);
    }

    if (!this.isUrlProtected(props) && includes(props.location.pathname, 'auth/register') && this.hasAdminUser(props) && !skipAction) {
      this.props.history.push('/plugins/users-permissions/auth/login');
    }

    if (props.hasUserPlugin && !this.isUrlProtected(props) && !includes(props.location.pathname, 'auth/register') && !this.hasAdminUser(props)) {
      this.props.history.push('/plugins/users-permissions/auth/register');
    }

    if (!props.hasUserPlugin || auth.getToken() && !this.state.hasAlreadyRegistereOtherPlugins) {
      map(omit(this.props.plugins.toJS(), ['users-permissions', 'email']), plugin => {
        switch (true) {
          case isFunction(plugin.bootstrap) && isFunction(plugin.pluginRequirements):
            plugin.pluginRequirements(plugin)
              .then(plugin => {
                return plugin.bootstrap(plugin);
              })
              .then(plugin => this.props.pluginLoaded(plugin));
            break;
          case isFunction(plugin.pluginRequirements):
            plugin.pluginRequirements(plugin).then(plugin => this.props.pluginLoaded(plugin));
            break;
          case isFunction(plugin.bootstrap):
            plugin.bootstrap(plugin).then(plugin => this.props.pluginLoaded(plugin));
            break;
          default:
        }
      });

      this.setState({ hasAlreadyRegistereOtherPlugins: true });
    }
  }

  hasUserPluginLoaded = (props) => typeof get(props.plugins.toJS(), ['users-permissions', 'hasAdminUser']) !== 'undefined';

  hasAdminUser = (props) => get(props.plugins.toJS(), ['users-permissions', 'hasAdminUser']);

  isUrlProtected = (props) => !includes(props.location.pathname, get(props.plugins.toJS(), ['users-permissions', 'nonProtectedUrl']));

  shouldDisplayLogout = () => auth.getToken() && this.props.hasUserPlugin && this.isUrlProtected(this.props);

  showLeftMenu = () => !includes(this.props.location.pathname, 'users-permissions/auth/');

  retrievePlugins = () => {
    const { adminPage: { currentEnvironment }, plugins } = this.props;

    if (currentEnvironment === 'production') {
      let pluginsToDisplay = plugins;
      PLUGINS_TO_BLOCK_PRODUCTION.map(plugin =>
        pluginsToDisplay = pluginsToDisplay.delete(plugin));

      return pluginsToDisplay;
    }

    return plugins;
  }

  render() {
    const { adminPage } = this.props;
    const header = this.showLeftMenu() ? <Header /> : '';
    const style = this.showLeftMenu() ? {} : { width: '100%' };

    return (
      <div className={styles.adminPage}>
        {this.showLeftMenu() && (
          <LeftMenu
            plugins={this.retrievePlugins()}
            layout={adminPage.layout}
            version={adminPage.strapiVersion}
          />
        )}
        <CTAWrapper>
          {this.shouldDisplayLogout() && <Logout />}
          <LocaleToggle isLogged={this.shouldDisplayLogout() === true} />
        </CTAWrapper>
        <div className={styles.adminPageRightWrapper} style={style}>
          {header}
          <Content {...this.props} showLeftMenu={this.showLeftMenu()}>
            <Switch>
              <Route path="/" component={HomePage} exact />
              <Route path="/plugins/:pluginId" component={PluginPage} />
              <Route path="/plugins" component={ComingSoonPage} />
              <Route path="/list-plugins" component={ListPluginsPage} exact />
              <Route path="/install-plugin" component={InstallPluginPage} exact />
              <Route path="/configuration" component={ComingSoonPage} exact />
              <Route path="" component={NotFoundPage} />
              <Route path="404" component={NotFoundPage} />
            </Switch>
          </Content>
        </div>
        <OverlayBlocker isOpen={this.props.blockApp && this.props.showGlobalAppBlocker} />
      </div>
    );
  }
}

AdminPage.childContextTypes = {
  disableGlobalOverlayBlocker: PropTypes.func,
  enableGlobalOverlayBlocker: PropTypes.func,
  plugins: PropTypes.object,
  updatePlugin: PropTypes.func,
};

AdminPage.contextTypes = {
  router: PropTypes.object.isRequired,
};

AdminPage.defaultProps = {
  adminPage: {},
  hasUserPlugin: true,
};

AdminPage.propTypes = {
  adminPage: PropTypes.object,
  blockApp: PropTypes.bool.isRequired,
  disableGlobalOverlayBlocker: PropTypes.func.isRequired,
  enableGlobalOverlayBlocker: PropTypes.func.isRequired,
  getGaStatus: PropTypes.func.isRequired,
  getLayout: PropTypes.func.isRequired,
  hasUserPlugin: PropTypes.bool,
  history: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  pluginLoaded: PropTypes.func.isRequired,
  plugins: PropTypes.object.isRequired,
  showGlobalAppBlocker: PropTypes.bool.isRequired,
  updatePlugin: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  adminPage: selectAdminPage(),
  blockApp: makeSelectBlockApp(),
  hasUserPlugin: selectHasUserPlugin(),
  plugins: selectPlugins(),
  showGlobalAppBlocker: makeSelectShowGlobalAppBlocker(),
});

function mapDispatchToProps(dispatch) {
  return {
    disableGlobalOverlayBlocker: () => { dispatch(disableGlobalOverlayBlocker()); },
    enableGlobalOverlayBlocker: () => { dispatch(enableGlobalOverlayBlocker()); },
    getGaStatus: () => { dispatch(getGaStatus()); },
    getLayout: () => { dispatch(getLayout()); },
    onHideNotification: (id) => { dispatch(hideNotification(id)); },
    pluginLoaded: (plugin) => { dispatch(pluginLoaded(plugin)); },
    updatePlugin: (pluginId, updatedKey, updatedValue) => { dispatch(updatePlugin(pluginId, updatedKey, updatedValue)); },
    dispatch,
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);
const withReducer = injectReducer({ key: 'adminPage', reducer });
const withSaga = injectSaga({ key: 'adminPage', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(AdminPage);

// export default connect(mapStateToProps, mapDispatchToProps)(AdminPage);
```

then remove the analytics plugin: 
```
cd my-app && rm -rf plugins/analytics
```

Then at the root of this new app run `npm run setup` 

Once the setup is complete copy the admin folder into your other app and it should be working.

If you encounter any problem you can DM on slack

**Comment by israelroldan** [comment]
For me removing the analytics plugin using the admin page fixed this issue!

**Comment by dsheyp** [comment]
exactly the same issue here.

my understanding of the problem is the following: when you create the new (admin) user a cookie will be set in the browser. but if this cookie is deleted later in the browser, strapi will always redirect that browser to the "register" page and it will be impossible to re-login with that browser until you create a new account from that browser which will again set the cookie.

this needs to be more robust, may be by completely dropping the automatic redirect to the register page and just let users decide themselves whether they want to go to "singup/register" or "login" by letting them cklick on the respective link.

**Comment by dsheyp** [comment]
@soupette

i applied your fix but it doesn't change anything for me when there is no cookie present in the browser - the browser will always be redirected to the register page although there exists an admin user in the mongodb. with a different browser where the cookie is still present i can login at the same time using exactly that account. i think it's the missing cookie in the browser (on which strapi relies) which triggers that issue. may be the same thing also happens after upgrading strapi: the cookie isn't valid anymore after upgrade, therefore the browser will be redirected to the "register" page.
